tests/compile-fail/while_loop.rs

```diff
@@ -0,0 +1,27 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#[deny(while_let_loop)]
+fn main() {
+    let y = Some(true);
+    loop { //~ERROR
+        if let Some(_x) = y {
+            let _v = 1;
+        } else {
+            break;
+        }
+    }
+    loop { //~ERROR
+        if let Some(_x) = y {
+            let _v = 1;
+        } else {
+            break
+        }
+    }
+    loop { //~ERROR
+        match y {
+            Some(_x) => true,
+            None => break
+        };
+    }
+}
```

Comment:
Without else? That shouldn't get matched, but I can add it as a test case of course.